[BUG] build: source-release RAT validation fails on hudi-trino/.mvn/modernizer files

### Bug Description

**What happened:**

`validate_staged_release.sh` runs `validate_source_rat.sh` inside the extracted source tarball, and that check fails on master:

```
Rat check: Summary over all files. Unapproved: 2, unknown: 2, generated: 0, approved: 5609 licenses.
[WARNING] Files with unapproved licenses:
  hudi-trino/.mvn/modernizer/violations-production-code-only.xml
  hudi-trino/.mvn/modernizer/violations.xml
[ERROR] Too many files with unapproved license: 2
```

Both files are tracked and carry no ASF header. The root pom's `<exclude>.mvn/**</exclude>` does not cover them - a RAT exclude is relative to the project basedir, so it only ever matches `<root>/.mvn/**`. In the workspace they are hidden by `.gitignore`, which apache-rat 0.16.1 reads, but `create_source_directory.sh` strips `.gitignore` from the source release.

**What you expected:**

`validate_staged_release.sh` to pass on a source tarball built from master.

**Steps to reproduce:**

```
./scripts/release/create_source_directory.sh /tmp/hudi-src
cd /tmp/hudi-src && mvn apache-rat:check -DdeployArtifacts=true
```

| tree | result |
|---|---|
| workspace (`.gitignore` present) | `Unapproved: 0`, BUILD SUCCESS |
| source release directory | `Unapproved: 2`, BUILD FAILURE |

### Environment

**Hudi version:** master (1.3.0-SNAPSHOT)
**apache-rat-plugin version:** 0.16.1

### Logs and Stack Trace

```
[ERROR] Failed to execute goal org.apache.rat:apache-rat-plugin:0.16.1:check (default-cli) on project hudi: Too many files with unapproved license: 2 See RAT report in: /tmp/hudi-src/target/rat.txt
```

--- a/pom.xml
+++ b/pom.xml
@@ -706,7 +706,9 @@
               <exclude>**/banner.txt</exclude>
               <!-- local files not in version control -->
               <exclude>**/*.iml</exclude>
-              <exclude>.mvn/**</exclude>
+              <!-- .mvn at any depth: nested ones are tracked without an ASF header
+                   (hudi-trino/.mvn/modernizer/*.xml) and the source release strips .gitignore. -->
+              <exclude>**/.mvn/**</exclude>
             </excludes>
           </configuration>
           <executions>

